atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderAppender.java

```diff
@@ -735,7 +742,7 @@ else if (member.canAppend()) {
 
   private boolean hasMoreEntries(final RaftMemberContext member) {
     // If the member's nextIndex is an entry in the local log then more entries can be sent.
-    return !member.hasReplicationContext() || member.hasNextEntry();
+    return member.hasNextEntry();
```

Comment:
:question: What if here reader is null because replication context is closed?

Reply:
I'll just remove this change. I think we don't run into this case anymore but it doesn't hurt to prevent an ugly NPE :+1: